ntoskrnl/include/internal/i386/ke.h

```diff
@@ -346,8 +347,9 @@ KeQueryInterruptHandler(IN ULONG Vector)
     //
     // Read the entry from the IDT
     //
-    return (PVOID)(((Pcr->IDT[Entry].ExtendedOffset << 16) & 0xFFFF0000) |
-                    (Pcr->IDT[Entry].Offset & 0xFFFF));
+    Handler = (ULONG_PTR)(USHORT)Pcr->IDT[Entry].ExtendedOffset << 16;
+    Handler |= (ULONG_PTR)(USHORT)Pcr->IDT[Entry].Offset;
```

Comment:
What was the actual fix kasan claimed was needed here?

Reply:
> UBSan: Undefined Behavior in ../ntoskrnl/include/internal/i386/ke.h:349:53, left shift of 32832 by 16 places cannot be represented in type 'int'